### ISSUE TITLE
HazelcastClient (3.4.1) not importing in OSGi environment

### ISSUE THREAD

**Issue by cogtinuum** [hazelcast/hazelcast#4782]
I'm using Hazelcast 3.4.1 on an equinox OSGi runtime. My target environment (PDE) contains the hazelcast-3.4.1.jar bundle and its concomitant fragment hazelcast-client-3.4.1.jar. My personal bundle manifest looks as follows:
```
Import-Package: com.hazelcast.client;version="3.4.1",
 com.hazelcast.client.config;version="3.4.1",
 com.hazelcast.config;version="3.4.1",
 com.hazelcast.core;version="3.4.1",
 ...
```
Within a class inside my personal bundle I'm importing as follows:
```
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.client.config.ClientConfig;
```
The PDE will not recognize the classes HazelcastClient or ClientConfig within the com.hazelcast.client or com.hazlecast.client.config packages, respectively. The hazelcast jar MANIFEST's make extensive use of the "uses" directive so I'm wondering if there's collisions occurring during resolution.

When I launch the OSGi runtime, the Hazelcast bundle and the client fragment will load:
```
9   ACTIVE      com.hazelcast_3.4.1
            Fragments=10
10  RESOLVED    com.hazelcast.client_3.4.1
            Master=9
```
If I run the headers command on bundle 9 I get:
```
osgi> headers 9
  ...  
  Export-Package = com.hazelcast.client;uses:="c...
  ...
``` 
If I run the packages command on bundle 9 I get:
```
osgi> packages 9
No exported packages
```
So it is clearly exported but not recognized as an export.  I'm using Eclipse Luna SR2.
Any suggestions that might help me resolve this would be greatly appreciated.

**Comment by noctarius** [comment]
thanks for creating the issue, we'll look into it! :)

**Comment by cogtinuum** [comment]
Thank you noctarius!

**Comment by serkan-ozal** [comment]
Hi @cogtinuum 

`packages` commands works with `<package_name>` not with `<bundle_id>`.

So can you try with `packages com.hazelcast`?

**Comment by serkan-ozal** [comment]
`packages` command is used to list all the bundles which use the given package.
`packages com.hazelcast.core` is works for me. So there is no issue about both of Hazelcast and OSGI

**Comment by cogtinuum** [comment]
@serkan-ozal thank you for looking into my original issue.  You are correct in stating that there was no issue with Hazelcast or OSGi.  I've since found the solution to my original problem and felt obligated to post it here so someone else doesn't bother the group with the same problem.

The Hazelcast client api is a fragment of the core Hazelcast bundle.  If you're using the Eclipse PDE platform to manage OSGi development and deployment then, by default, the PDE DOES NOT expose api contributed by fragment bundles.  Ergo, the Hazelcast client code will not show up in the manifest "dependencies" area and the PDE will generally throw resolution errors at every use of fragment api.  This does not effect runtime OSGi code, only the PDE development area.  The following link has a bit more background on this:

[Eclipse export API from fragment](http://stackoverflow.com/questions/23683708/osgi-eclipse-how-to-export-package-from-a-fragment-bundle-and-make-them-visible)

To get around this one needs to augment the manifest within the hazelcast core jar with the following line:

`Eclipse-ExtensibleAPI: true`

I created a manifest-hz-mod.mf with that line in it (ensure the file ends with a newline or carriage return) and updated the manifest as follows:

`jar ufm hazelcast-3.6.jar manifest-hz-mod.mf`

To check the mod, I dumped the manifest as follows:

`unzip -p hazelcast-3.6.jar META-INF/MANIFEST.MF | grep -in eclipse`

Once done, the PDE target can be reloaded with the modified library and all of the Hazelcast client api will be available in the PDE environment.

Again, thanks for taking the time to research this originally.
